### PR TITLE
Add Babel setup

### PR DESCRIPTION
**Pull request by pmuens**
## What has been implemented?

Adds a Babel setup for the components core and all components in the registry. Support is added for Node versions as early as 4. Travis and AppVeyor build matrices were updated as well.

Closes SC-16

## Steps to verify

Run `npm run build` and / or `npm run watch`.

## Todos:

* [x] Write / update documentation
* [x] Run Prettier

### CODE REVIEW

**Comment by brianneisler** [comment]
@pmuens, I wonder if it would be more maintainable to have one higher level build setup for all of the components instead of a separate one for each. Each component needs its own `package.json` to  include its own dependencies during installation but i'm not sure if each one needs a separate build setup. Any thoughts on when we might need a setup like that?

**Comment by pmuens** [comment]
> @pmuens, I wonder if it would be more maintainable to have one higher level build setup for all of the components instead of a separate one for each. Each component needs its own package.json to include its own dependencies during installation but i'm not sure if each one needs a separate build setup. Any thoughts on when we might need a setup like that?

Thank you for your review @brianneisler 👍

Yes, I actually wrote a bash script while working on this to add the scripts and the dependencies automatically since doing this manually for every component would've been a real pain.

I thought along those lines but after thinking more about it I came up with the conclusion that every component should be self-contained since they might reside in other locations later on. Maybe that's a premature optimiation though. I'm open to other thoughts as well.

File: registry/build-components.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      '**/node_modules',
+      '--ignore',
+      '**/*.test.js'
+    ],
+    { env: process.env, cwd: componentDirPath }
+  )
+  command.stdout.on('data', (data) => {
+    console.log(data.toString())
+  })
+  command.stdout.on('close', () => BbPromise.resolve())
```
[the remark this answers is not in the document]
Comment (pmuens) [reply registry/build-components.js]: D'oh 🤦‍♂️ what a stupid mistake! Thanks for spotting @brianneisler 👍 🙏 💯

**Comment by pmuens** [comment]
After spending a good portion of time on this PR we can finally see that all builds for all the supported Node.js versions (4 - 10) are finally green on Travis and AppVeyor.

This PR can be considered done and needs a final review so that we can merge it :shipit:

/cc @eahefnawy @brianneisler

**Comment by eahefnawy** [comment]
imo we should spend some time manually testing this PR in different node environments before merging. Travis can sometimes be misleading.

**Comment by pmuens** [comment]
> @pmuens Is there a specific reason we are holding off on merging this PR? Let me know.

@ganeshvlrk no. No specific reason. I haven't tested the support for all the different Node.js versions (this would be a daunting task). IMHO seeing that the tests pass should be enough for now...

**Comment by ganeshvlrk** [comment]
Agreed. Let's go ahead and merge this on the basis  of the tests passing

**Comment by pmuens** [comment]
> Agreed. Let's go ahead and merge this on the basis of the tests passing

@ganeshvlrk sounds good :+1:

Can I just need an approval and we should be ready to merge this /cc @eahefnawy @brianneisler

**Comment by pmuens** [comment]
Just tested this today running it in a fresh Node 4 Docker container.

Had to make some slight changes (e.g. Babel should be added to `dependencies` rather than `devDependencies` so that it's available when the package is installed globally). After that everything worked like a charm. Would like to merge it after the build passes 🚢 :shipit:

**Pull request merged**